sw/device/tests/alert_handler_lpg_sleep_mode_alerts.c

```diff
@@ -0,0 +1,528 @@
+ */
+static void init_peripherals(void) {
+  mmio_region_t base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_plic_init(base_addr, &plic));
+
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR);
+  CHECK_DIF_OK(dif_alert_handler_init(base_addr, &alert_handler));
+
+  // Initialize pwrmgr
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+
+  // Initialize aon_timer
+  CHECK_DIF_OK(dif_aon_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR), &aon_timer));
+
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+
+  mmio_region_t ibex_addr =
+      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_core_ibex_init(ibex_addr, &ibex));
+
+  CHECK_DIF_OK(dif_flash_ctrl_init_state(
+      &flash_ctrl,
+      mmio_region_from_addr(TOP_EARLGREY_FLASH_CTRL_CORE_BASE_ADDR)));
+}
+
+// NVM counters/fields to keep test steps between deep sleep modes
+enum {
+  // The counter ID for the non-volatile counter keeping the test steps.
+  kCounterTestSteps = 0,
+  // The counter ID for the non-volatile counter keeping the number of max
+  // wakeup events
+  kCounterMaxWakeups = 1,
+};
+
+/**
+ * Program the alert handler to escalate on alerts upto phase 1 (i.e. wipe
+ * secret) but not trigger reset. Then CPU can check if the correct interrupt
+ * fires and check the local alert cause register.
+ */
+static void alert_handler_config(uint32_t ping_timeout) {
+  dif_alert_handler_alert_t alerts[ALERT_HANDLER_PARAM_N_ALERTS];
+  dif_alert_handler_class_t alert_classes[ALERT_HANDLER_PARAM_N_ALERTS];
+
+  // Enable all incoming alerts and configure them to classa.
+  // This alert should never fire because we do not expect any incoming alerts.
+  for (int i = 0; i < ALERT_HANDLER_PARAM_N_ALERTS; ++i) {
+    alerts[i] = i;
+    alert_classes[i] = kDifAlertHandlerClassA;
+  }
+
+  // Enable alert ping fail local alert and configure that to classb.
+  // Enable other local alerts and configure them to classa.
+  dif_alert_handler_local_alert_t loc_alerts[] = {
+      kDifAlertHandlerLocalAlertAlertPingFail,
+      kDifAlertHandlerLocalAlertAlertIntegrityFail,
+      kDifAlertHandlerLocalAlertBusIntegrityFail,
+      kDifAlertHandlerLocalAlertEscalationIntegrityFail,
+      kDifAlertHandlerLocalAlertEscalationPingFail,
+      kDifAlertHandlerLocalAlertShadowedStorageError,
+      kDifAlertHandlerLocalAlertShadowedUpdateError};
+  dif_alert_handler_class_t loc_alert_classes[] = {
+      kDifAlertHandlerClassB, kDifAlertHandlerClassA, kDifAlertHandlerClassA,
+      kDifAlertHandlerClassA, kDifAlertHandlerClassA, kDifAlertHandlerClassA,
+      kDifAlertHandlerClassA};
+
+  dif_alert_handler_escalation_phase_t esc_phases[] = {
+      {.phase = kDifAlertHandlerClassStatePhase0,
+       .signal = 0,
+       .duration_cycles = 2000}};
```

Comment:
This delay seems excessive: 2000 aon cycles is 10 ms! The duration should correspond to what you expect to do during this phase.

Reply:
@matutem Thanks, I fixed both the irq_deadline and escalation duration values in the latest version of the code.